reviewrot/base.py

```diff
@@ -0,0 +1,101 @@
+from collections import OrderedDict
+
+
+class BaseService(object):
+
+    def format_duration(self, rel_diff):
+        """
+        Formats the duration the review request is pending for
+        Args:
+            rel_diff (relativedelta): duration the review request is pending
+                                      for
+
+        Returns:
+            a string of duration the review request is pending for
+        """
+        time_dict = OrderedDict([
+            ('year', rel_diff.years),
+            ('month', rel_diff.months),
+            ('day', rel_diff.days),
+            ('hour', rel_diff.hours),
+            ('minute', rel_diff.minutes),
+        ])
+
+        result = []
+        for k, v in time_dict.items():
+            if v == 1:
+                result.append('%s %s' % (v, k))
+            elif v > 1:
+                result.append('%s %ss' % (v, k))
+
+        return ' '.join(result)
+
+    def check_request_state(self, abs_diff, rel_diff, state_, value, duration):
+        """
+        Checks if the review request is older or newer than specified
+        time interval.
+        Args:
+            abs_diff (relativedelta): absolute time difference between now and
+                                      review request filed
+
+            rel_diff (str): relative time difference between now and
```

Comment:
I don't think `rel_diff` is a string based on the code. As a result, I'm a little confused about the differences between abs_diff and rel_diff if they are both relativedelta objects.